### PR TITLE
[Experimental] Product Filters Redesign > Status filter block: Rename Stock Status Filter to Status Filter

### LINKED ISSUE #51859: Rename Stock filter block to Status
The Stock filter has a new name: Status. We need to update the naming to the new ones for these following places:
- Block title.
- Block description.
- Block name (slug).
- Folder name.
- Tests and any other references in source code.

### PR DESCRIPTION
**Pull request by thealexandrelara**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #51859 

This PR introduces the following updates related to renaming the “Stock” filter to “Status” filter:

1. Block Title and Description updated: The block’s title and description now reflect the new “Status” name.
2. Slug Renaming: The block’s slug has been updated from “product-filter-stock-status” to “product-filter-status”.
3. Folder Name: The folder containing the block files has been renamed to align with the updated block name.
4. Icon Name and Webpack Entry: Both the block’s icon and Webpack entry point have been renamed to reflect the new name.
5. Source Code References: All references to the “Stock Status” filter in the source code have been updated to "Status" filter.
6. Test Updates: An E2E test that was failing after the updates has been fixed.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).

1. Navigate to Product Filters template (Appearance > Editor > Patterns > General > Product Filters (Experimental).
2. Click on it to open the template part editor.
3. Change to the Code Editor mode;
4. Replace the current code with the following:

```
<!-- wp:woocommerce/product-filters -->
<div class="wp-block-woocommerce-product-filters wc-block-product-filters"><!-- wp:heading {"level":3} -->
<h3 class="wp-block-heading">Status Filter</h3>
<!-- /wp:heading -->

<!-- wp:woocommerce/product-filter-status /--></div>
<!-- /wp:woocommerce/product-filters -->
```

5. Make sure that you can see the Status filter being rendered in the canvas.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment  Rename the Stock Status filter block to Status block

</details>

### CODE REVIEW

**Comment by dinhtungdu** [comment]
I think we should wait for merging #51638 before this PR or we have to deal with pretty big merge conflicts. Luigi also opened #51856 to rename the block afterward fyi.

**Review by roykho — APPROVED** [review]
Changes looks good. Good job!

✅ I see the updated block title and description
✅ I see the updated folder names

Not blocking but I am wondering if we should also change the variable names throughout the code for example https://github.com/woocommerce/woocommerce/blob/9ff2d43ecfab387b51a52b8686af65b5b57a3692/plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/edit.tsx#L26 and change it to `StatusCount` to be consistent. But other than that I agree with Tung to wait for https://github.com/woocommerce/woocommerce/pull/51714 to merge.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/block.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-	"name": "woocommerce/product-filter-stock-status",
+	"name": "woocommerce/product-filter-status",
 	"version": "1.0.0",
-	"title": "Filter Options",
+	"title": "Status",
```
Comment (retrofox) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/block.json]:
```suggestion
	"title": "Status (Experimental)",
```

?

Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/block.json]: Nice catch! Done in 95afca1c9192684567cfd4a2ba69eb25e71bd272

**Comment by thealexandrelara** [comment]
> I think we should wait for merging #51638 before this PR or we have to deal with pretty big merge conflicts. Luigi also opened #51856 to rename the block afterward fyi.

I agree, let's wait for that PR to be merged and them we can merge this one.

> Not blocking but I am wondering if we should also change the variable names throughout the code for example and change it to StatusCount to be consistent. 

Nice catch! I just updated it on d4e55486400bbc89b4b6c2c641d19242d19236c3

> But other than that I agree with Tung to wait for https://github.com/woocommerce/woocommerce/pull/51714 to merge.

Yeah, let's agree for that to be merged.

**Pull request merged**